docs/examples/create_and_run_flow/manage_flow.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+import argparse
+import os
+import sys
+import time
+
+import globus_sdk
+from globus_sdk.tokenstorage import SimpleJSONFileAdapter
+
+MY_FILE_ADAPTER = SimpleJSONFileAdapter(os.path.expanduser("~/.sdk-manage-flow.json"))
+
+# tutorial client ID
+# we recommend replacing this with your own client for any production use-cases
+CLIENT_ID = "61338d24-54d5-408f-a10d-66c06b59f6d2"
+
+NATIVE_CLIENT = globus_sdk.NativeAppAuthClient(CLIENT_ID)
+
+
+def do_login_flow(scope):
+    NATIVE_CLIENT.oauth2_start_flow(requested_scopes=scope)
+    authorize_url = NATIVE_CLIENT.oauth2_get_authorize_url()
+    print(f"Please go to this URL and login:\n\n{authorize_url}\n")
+    auth_code = input("Please enter the code here: ").strip()
+    tokens = NATIVE_CLIENT.oauth2_exchange_code_for_tokens(auth_code)
+    return tokens
+
+
+def get_tokens(flow_id=None, force=False):
+    if flow_id:
+        resource_server = flow_id
+        scope = globus_sdk.SpecificFlowClient(flow_id).scopes.user
+    else:
+        resource_server = globus_sdk.FlowsClient.resource_server
+        scope = globus_sdk.FlowsClient.scopes.manage_flows
+
+    # try to load the tokens from the file, possibly returning None
+    if MY_FILE_ADAPTER.file_exists():
+        tokens = MY_FILE_ADAPTER.get_token_data(resource_server)
+    else:
+        tokens = None
+
+    if tokens is None or force:
+        # do a login flow, getting back initial tokens
+        response = do_login_flow(scope)
+        # now store the tokens and pull out the correct token
+        MY_FILE_ADAPTER.store(response)
+        tokens = response.by_resource_server[resource_server]
+
+    # allow for 60 seconds of potential clock skew or delay
+    # if the call was not "force=True", potentially re-enter
+    leeway = 60
+    if not force and (time.time() > tokens["expires_at_seconds"] - leeway):
+        tokens = get_tokens(flow_id, force=True)
```

Comment:
I'm going to switch these to using refresh tokens to address this concern.
It is somewhat important that an application which persists access tokens does some work to invalidate or replace those tokens when they expire. But the problem has a tidier, simpler solution in the refresh token case because the validate-and-replace logic is baked into the authorizer object.